### PR TITLE
nih example updates

### PR DESCRIPTION
**Pull request by RadhakrishnaJ**
nih example initial updates

Cleanup, rework and test will continue in subsequent updates

### CODE REVIEW

File: examples/nih/README.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
+NIH
+========
+
+The purpose is this example is to show case Swarm usecase with real world NIH dataset [1]. NIH dataset is a well known health care dataset. NIH dataset has complex xray images, it is a challenge task to meet better metrics on this datset.
```
Comment (suresh-ls) [inline examples/nih/README.md:4]: is - > of

File: examples/nih/data/data_generator.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+######################################################################
+# (C)Copyright 2024 Hewlett Packard Enterprise Development LP
+######################################################################
+yamlfilename  : nih_data_generator
+inputpathofnihdata : /home/radhakrishnaj/Radha/NIHRAWDATA/input
```
Comment (suresh-ls) [inline examples/nih/data/data_generator.yaml:5]: Can remove usernames and make it generic, say swarm_user

File: examples/nih/README.md [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+
+The data processing and model code for this example has been taken from multiple kaggle kernels and modified to run on a Swarm Learning platform.
+
+This example uses node specific biased data and one common test data. The Machine Learning program, after conversion to Swarm Learning for the TensorFlow-based Keras platform, is in `examples/nih/model`. The TensorFlow-based file is called `nih_tf.py`.
+
+This example shows the Swarm training of NIH model using two Machine Learning (ML) nodes. Machine Learning nodes are automatically spawned by Swarm Operators (SWOP) nodes running on two different hosts. Swarm training is initiated by Swarm Command Interface (SWCI) node and orchestrated by two Swarm Network (SN) nodes running in different hosts. This example also shows how private data and shared model can be mounted to Machine Learning nodes for Swarm training. For details, see the profile files and task definition files placed under `examples/nih/swop` and `examples/nih/swci` folders respectively.
```
Comment (suresh-ls) [inline examples/nih/README.md:10]: 2 or 3 nodes ?

File: examples/nih/README.md [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+
+
+
+## Cluster Setup
+
+The cluster setup for this example uses 2 hosts, as shown in the figure below:
```
Comment (suresh-ls) [inline examples/nih/README.md:16]: 2 -> 1 host

File: examples/nih/README.md [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+
+
+1. This example uses one Swarm Network (SN) node. The names of the docker containers representing these two nodes are **sn1** . sn1 is the Sentinel Node. sn1 runs on host 172.1.1.1. 
+2. Swarm Learning (SL) and Machine Learning (ML) nodes are automatically spawned by Swarm Operators (SWOP) nodes during training and removed after training. Example one SWOP node named **swop1**. swop1 runs on host 172.1.1.1.
+3. Training is initiated by SWCI node (**swci1**) that runs on host 172.1.1.1
+4. Example assumes that License Server already runs on host 172.1.1.1. All Swarm nodes connect to the License Server, on its default port 5814.
+
+
+
+## Running the MNIST example
```
Comment (suresh-ls) [inline examples/nih/README.md:27]: MNIST -> NIH

File: examples/nih/README.md [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+    
+    Note: If required, modify IP, according to environment, in `workspace/mnist/swci/swci-init` file.  
+   ```
+./scripts/bin/run-swci --rm --name=swci1 --network=host-1-net --usr-dir=workspace/nih/swci --init-script-name=swci-init       \
+--key=workspace/nih/cert/swci-1-key.pem --cert=workspace/nih/cert/swci-1-cert.pem --capath=workspace/nih/cert/ca/capath   \
+-e http_proxy= -e https_proxy= --apls-ip=${APLS_IP}
+
+   ```
+
+12.	Three nodes Swarm training is automatically started when the run task gets assigned and executed. User can open a new terminal on both host-1 and monitor the docker logs of ML nodes for Swarm training. Swarm training will end with the following log message at the end –
```
Comment (suresh-ls) [inline examples/nih/README.md:128]: both host-1 -> host-1

File: examples/nih/README.md [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+   ```
+./scripts/bin/run-swci --rm --name=swci1 --network=host-1-net --usr-dir=workspace/nih/swci --init-script-name=swci-init       \
+--key=workspace/nih/cert/swci-1-key.pem --cert=workspace/nih/cert/swci-1-cert.pem --capath=workspace/nih/cert/ca/capath   \
+-e http_proxy= -e https_proxy= --apls-ip=${APLS_IP}
+
+   ```
+
+12.	Three nodes Swarm training is automatically started when the run task gets assigned and executed. User can open a new terminal on both host-1 and monitor the docker logs of ML nodes for Swarm training. Swarm training will end with the following log message at the end –  
+    `SwarmCallback : INFO : All peers and Swarm training rounds finished. Final Swarm model was loaded.`  
+    Final Swarm model will be saved inside \<PROJECT\> location which will likely be user's specific as `workspace/mnist/<userN>` directory on both the hosts. All the dynamically spawned SL and ML nodes will exit after Swarm training. The SN and SWOP nodes continue running.
```
Comment (suresh-ls) [inline examples/nih/README.md:130]: mnist -> nih

2) Please do a grep of mnist

File: examples/nih/swci/swci-init [hunk 8]
```diff
@@ -0,0 +1,77 @@
+######################################################################
+# (C)Copyright 2021-2023 Hewlett Packard Enterprise Development LP
```
Comment (suresh-ls) [inline examples/nih/swci/swci-init:2]: copyright 2024

File: examples/nih/README.md [hunk 9]
```diff
@@ -0,0 +1,138 @@
+NIH
+========
+
+The purpose is this example is to show case Swarm usecase with real world NIH dataset [1]. NIH dataset is a well known health care dataset. NIH dataset has complex xray images, it is a challenge task to meet better metrics on this datset.
```
Comment (suresh-ls) [inline examples/nih/README.md:4]: challenge -> challenging

**Review by girisairam450 — APPROVED** [review]
Reviewed

File: examples/nih/model/nih_tf.py [hunk 10]
```diff
@@ -0,0 +1,298 @@
...
+##
+## Unless required by applicable law or agreed to in writing, software
+## distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+## WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+## License for the specific language governing permissions and limitations
+## under the License.
+############################################################################
+
+
+# Some the code are being used from below kaggle kernel
```
Comment (suresh-ls) [inline examples/nih/model/nih_tf.py:17]: Change this to "inspired.. " like the other file comment

File: examples/nih/data/nih_nodes_data_processing.py [hunk 11]
```diff
@@ -0,0 +1,198 @@
...
+#count_per_unique_label = xray_data['Finding Labels'].value_counts() # get frequency counts per label
+#df_count_per_unique_label = count_per_unique_label.to_frame() # convert series to dataframe for plotting purposes
+df_count_per_unique_label = xray_data['Finding Labels'].value_counts().reset_index().rename(columns={'index': 'Finding Labels', 0: 'count'})
+
+print(df_count_per_unique_label)
+
+sns.barplot(x = 'Finding Labels' , y="count", data=df_count_per_unique_label[:20], color = "green"), plt.xticks(rotation = 90) # visualize results graphically
+
+
+# define labels of our ineterest
```
Comment (suresh-ls) [inline examples/nih/data/nih_nodes_data_processing.py:104]: ineterest -> interest

**Review by suresh-ls — APPROVED** [review]

**Pull request merged**